wfe2/wfe.go

```diff
@@ -1416,8 +1416,10 @@ func (wfe *WebFrontEndImpl) Account(
 		return
 	}
 
-	// If the body was not empty, then this is an account update request.
-	if string(body) != "" {
+	// If the body was not either completely empty or an empty JSON object, then
+	// this is an account update request. Treating the empty JSON object like a
+	// POST-as-GET is a holdover from ACMEv1.
+	if string(body) != "" && string(body) != "{}" {
```

Comment:
Because this is so surprising, and not directly related to your refactor except that your refactor helped us catch this bug, I'd break this four-line change out into its own PR that we land first. That way we can very thoroughly document it in the PR description and make sure everyone on the team understands exactly what's happening here.

Reply:
Agreed, thanks! I was committing it here in order to play with test fixes. Made #7844